pkg/buildscript/buildscript.go

```diff
@@ -41,22 +44,43 @@ func New() *BuildScript {
 	return bs
 }
 
+func (b *BuildScript) Project() string {
+	return b.project
+}
+
+func (b *BuildScript) SetProject(url string) {
+	b.project = url
+}
+
 func (b *BuildScript) AtTime() *time.Time {
-	return b.raw.AtTime
+	return b.atTime
 }
 
 func (b *BuildScript) SetAtTime(t time.Time) {
-	b.raw.AtTime = &t
+	b.atTime = &t
 }
 
 func (b *BuildScript) Equals(other *BuildScript) (bool, error) {
-	myBytes, err := b.Marshal()
+	b2, err := b.Clone()
+	if err != nil {
+		return false, errs.Wrap(err, "Unable to clone buildscript")
+	}
+	other2, err := other.Clone()
+	if err != nil {
+		return false, errs.Wrap(err, "Unable to clone other buildscript")
+	}
+
+	// Do not compare project URLs.
```

Comment:
If we compare URLs (which include commit IDs), we'll more often than not get false "your buildscript has changes, please commit" errors. Just compare the build expression and at time bits of the build script.